fix MethodError for connect(addr::Base.InetAddr{T<:Base.IPAddr})

mmaxs opened the pull request:
There is public method for the connect() function with the following signature:
`connect(addr::Base.InetAddr{T<:Base.IPAddr})`
It throws

```
MethodError: `connect!` has no method matching connect!(::TCPSocket, ::Base.InetAddr{...})
```

Here is the test example:

```
using Base.Test

# test method matching connect!(::TCPSocket, ::Base.InetAddr{T<:Base.IPAddr})
@test let
  addr = Base.InetAddr(ip"127.0.0.1", 4444)
  srv = listen(addr)
  @async try c = accept(srv); close(c) catch end
  local t = TCPSocket(), t0 = t
  try
    t = connect(addr)
  finally
    close(srv)
  end
  !is(t,t0)
end
```

PR fixes the issue. I've not included the above test since after the missing method has been added the test actually tests nothing but only the method existence.

tkelman commented:
Code shouldn't be added without a test that covers it, so do add some sort of test.

Commit:
fix MethodError for connect(addr::Base.InetAddr{T<:Base.IPAddr})

Without the fix, calling connect(addr::Base.InetAddr{T<:Base.IPAddr}) throws
MethodError: `connect!` has no method matching connect!(::TCPSocket, ::Base.InetAddr{...})

mmaxs commented:
OK

tkelman commented on test/socket.jl:
```diff
@@ -268,3 +268,34 @@ let
+
+# test the method matching connect!(::TCPSocket, ::Base.InetAddr{T<:Base.IPAddr})
+let
+  addr = Base.InetAddr(ip"127.0.0.1", 4444)
```
4 space indent
